R/qmd2md.R

```diff
@@ -1,50 +1,80 @@
-.qmd2md <- function(source_file, tar_dir) {
+.qmd2md <- function(source_file, tar_dir, verbose = FALSE) {
   if (missing(source_file) || !file.exists(source_file)) {
     stop("source_file must be a valid file path.", call. = FALSE)
   }
-
   if (!fs::dir_exists(tar_dir)) {
     stop("tar_dir must be a valid directory.", call. = FALSE)
   }
 
   tar_file <- fs::path_join(c(tar_dir, basename(source_file)))
   fs::file_copy(source_file, tar_file, overwrite = TRUE)
 
-  quarto::quarto_render(
-    input = path.expand(tar_file),
-    output_format = "md",
-    quiet = TRUE
-  )
+  if (isTRUE(verbose)) {
+    out <- try(quarto::quarto_render(
+      input = path.expand(tar_file),
+      output_format = "md",
+      quiet = FALSE
+    ), silent = FALSE)
+  } else {
+    void <- utils::capture.output(
+      out <- try(quarto::quarto_render(
+        input = path.expand(tar_file),
+        output_format = "md",
+        quiet = TRUE
+      ), silent = TRUE)
+    )
+  }
+
+
+  out <- ifelse(inherits(out, "try-error"), FALSE, TRUE)
+
+  return(out)
 }
 
 
-.rmd2md <- function(source_file, output_dir) {
+.rmd2md <- function(source_file, tar_dir, verbose = FALSE) {
   if (missing(source_file) || !file.exists(source_file)) {
     stop("source_file must be a valid file path.", call. = FALSE)
   }
+  if (!fs::dir_exists(tar_dir)) {
+    stop("tar_dir must be a valid directory.", call. = FALSE)
+  }
 
-  out <- utils::capture.output(
-    rmarkdown::render(
+  if (isTRUE(verbose)) {
+    out <- try(rmarkdown::render(
       input = path.expand(source_file),
-      output_format = "md_document",
-      quiet = TRUE,
+      output_format = "github_document",
+      quiet = !verbose,
       envir = new.env()
-    )
-  )
+    ), silent = FALSE)
+  } else {
+    void <- utils::capture.output({
```

Comment:
do you remember why this is needed? if it's not, this if condition can be refactored

Reply:
rmarkdown generates a *ton* of stuff in the console. I couldn't find another way to silence everything.